src/rust/src/arrow_interop/to_rust.rs

```diff
@@ -99,7 +99,7 @@ pub fn to_rust_df(rb: Robj) -> Result<pl::DataFrame, String> {
         }?;
 
         // no need to check as a record batch has the same guarantees
-        let df_res: Result<_, String> = Ok(DataFrame::new_no_checks(series_vec));
+        let df_res: Result<_, String> = Ok(DataFrame::new(series_vec).unwrap());
```

Comment:
@eitsupi I made this change, but cargo was suggesting instead to keep `new_no_checks` and to mark the function as `unsafe`. I'm not sure my choice here is the right one, do you know what's best?

Reply:
relevant PR: https://github.com/pola-rs/polars/pull/14443

Reply:
As noted in the above comment, I think this process is fine with `unsafe`.
I think now we can follow the IDE's suggestion and update it in case of problems.

Reply:
Ok I'll revert this one